### ISSUE TITLE
x/pkgsite: header structure on Details Page differs from tree outline

### ISSUE THREAD

**Issue by jba** [golang/go#41107]
On Package Details page,` <ul role="tree" aria-label="Outline">…</ul>` suggests that all headers within `<div class="js-docContent">` are structurally equal. Right now, most headers are `<h3>` and so are marked as subsections of Overview `<h2 tabindex="-1" id="pkg-overview" class="Documentation-overviewHeader">…</h2>`. If these are not subsections of overview, they should also be `<h2>`. Additionally, the `<ul role="tree" aria-label="Outline">…</ul> `suggests that there is a logical grouping for Functions and Types which is not reflected as headers. Functions and Types should be added as header, `<h2>` if equal to Overview, or `<h3>` if subsections of Overview. Individual Functions and Types in the package should be grouped as subsections of the new headers, with either `<h3>` or `<h4>` tags depending on the heading level of their parent section. For consistency with the UL, my recommendation is for Overview, Examples, Constants, Variables, Functions and Types to be `<h2>` and individual functions and types to be marked as `<h3>` subsections of their parent headers.
